### PR TITLE
StackClient: Implement JobCollection.get() method

### PR DESCRIPTION
**Pull request by gregorylegarec**
This method is actually used here https://github.com/cozy/cozy-libs/blob/master/packages/cozy-harvest-lib/src/models/konnector/KonnectorJobWatcher.js#L100, and throw an error.

This PR also suggest to abstract get methods pattern into an higher utility classed named Collection.

### CODE REVIEW

File: packages/cozy-stack-client/src/Collection.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * Utility class to abstract an regroup identical methods and logics for
+ * specific collections.
+ */
+export class Collection {
+  /**
+   * Utility method aimed to return only one document.
+   * @param  {Object}  stackClient
+   * @param  {String}  id              Document identifier
```
Comment (edas) [inline packages/cozy-stack-client/src/Collection.js]: Doc says get(stackclient, id, endpoint, normalise func, method)
Prototype says get(stackclient, endpoint, { normalize, method }) (and no id)

File: packages/cozy-stack-client/src/Collection.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+   * @param  {String}  endpoint        Stack endpoint
+   * @param  {Func}    normalize       Callback to normalize response data
+   * @param  {String}  method          HTTP method
+   * @return {Promise}                 Promise of JsonAPI response
```
Comment (edas) [inline packages/cozy-stack-client/src/Collection.js]: As the function is async, it is already infered that you return a promise. @return should declare what you return inside that promise

File: packages/cozy-stack-client/src/Collection.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * Handler for error response which return a empty value for "not found" error
+ * @param  {Error} error
+ * @param  {Array|Object}  [data=[]] Data to return in case of "not found" error
+ * @return {Object}        JsonAPI response with empty data in case of "not
+ * found" error.
+ */
+export const dontThrowNotFoundError = (error, data = []) => {
+  if (error.message.match(/not_found/)) {
+    return Array.isArray(data)
```
Comment (edas) [inline packages/cozy-stack-client/src/Collection.js]: This seems overly magic for me

Comment (gregorylegarec) [reply packages/cozy-stack-client/src/Collection.js]: Does the comment I added suits you ?

File: packages/cozy-stack-client/src/Collection.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Utility class to abstract an regroup identical methods and logics for
+ * specific collections.
+ */
+export class Collection {
+  /**
+   * Utility method aimed to return only one document.
+   * @param  {Object}  stackClient
+   * @param  {String}  endpoint        Stack endpoint
+   * @param  {Func}    normalize       Callback to normalize response data
```
Comment (edas) [inline packages/cozy-stack-client/src/Collection.js]:
```suggestion
   * @param  {Object}    options       Callback to normalize response data
   * @param  {Func}    options.normalize       Callback to normalize response data
```

(and so on)

**Comment by Crash--** [comment]
@gregorylegarec Can you finish this PR? :D

**Comment by gregorylegarec** [comment]
Merging as soon as Travis is green.

**Pull request merged**